projects/plugins/jetpack/extensions/shared/components/product-management-controls/use-open-block-sidebar.js

```diff
@@ -2,9 +2,13 @@
  * WordPress dependencies
  */
 import { store as blockEditorStore } from '@wordpress/block-editor';
-import { useDispatch } from '@wordpress/data';
+import { useDispatch, dispatch, select } from '@wordpress/data';
 import { getEditorType, SITE_EDITOR, WIDGET_EDITOR } from '../../get-editor-type';
 
+const closePublishSidebarIfNeeded = () =>
+	select( 'core/edit-post' ).isPublishSidebarOpened() &&
+	dispatch( 'core/edit-post' ).closePublishSidebar();
```

Comment:
I think we could move all of this inside the hook.

Also what do you think if we skipped the `isOpened` check and just dispatched the action anyway? If the sidebar is open, the state doesn't change and renders aren't triggered. 🤔 

```js
import { store as editPostStore } from '@wordpress/edit-post';

export default function useOpenBlockSidebar( clientId ) {
  const { closePublishSidebar } = useDispatch( editPostStore );
  // ...

  // if post editor...
  closePublishSidebar();
  enableComplementaryArea( 'core/edit-post', 'edit-post/block' );
}
```

Reply:
Also fun: the site editor has a pre-publish panel too, but I think it's handled by an internal state, which is not accessible from external plugins.

https://github.com/WordPress/gutenberg/blob/4830c2c2b95dc612b73efffbd3ed03cd93cd83fa/packages/edit-site/src/components/editor/index.js#L76-L87

I haven't double-checked `edit-widgets` so I can't be sure about it.

I'd say it's a bit of an annoying glitch, but pretty minor and not a blocker for this PR.

Reply:
Done. I've looked in to the site-editor and edit widgets and:
- for site editor the issue wasn't reproduced because the panel is closed on out of focus.
- for the edit widgets we don't display the post/pre publish panels.